backend/app/crud/training.py

```
@@ -22,8 +34,92 @@ async def create_training_log(db: AsyncSession, log: TrainingLogCreate, user_id:
     db.add(db_log)
     await db.commit()
     await db.refresh(db_log)
+
+    # 保存後の統計を取得
+    stats_after = await get_training_stats(db, user_id)
+    new_total, new_exercises = get_totals(stats_after)
+
+    unlocked_ids = await check_and_unlock_yucchin(db, user_id, old_total, new_total, old_exercises, new_exercises)
+    
+    # スキーマに合わせて返却するために属性を追加
+    setattr(db_log, 'unlocked_yucchin_types', unlocked_ids)
```

Ruff (0.14.8)

Do not call `setattr` with a constant attribute value. It is not any safer than normal property access.

Replace `setattr` with assignment

(B010)